File: api/app/schemas/friend.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

class FriendBase(BaseModel):
    name: str
    user_id: int

class FriendCreate(BaseModel):
    name: str
    user_id: Optional[int] = None

class FriendUpdate(BaseModel):
    name: str

class FriendInDB(FriendBase):
    id: int
    user_id: int

    class Config:
        orm_mode = True

class FriendAttributeResponse(BaseModel):
    attribute_name: str
    value: str

class FriendDetailRequest(BaseModel):
    friend_id: int

class ConversationHistoryItem(BaseModel):
    context: str
    conversation_date: datetime

class FriendDetailResponse(BaseModel):
    friend_name: str
    attributes: List[FriendAttributeResponse]
    conversations: List[ConversationHistoryItem]

class FriendAttributeUpdate(BaseModel):
    attribute_name: str
    value: str

class UpdateFriendDetailsRequest(BaseModel):
    friend_id: int
    attributes: List[FriendAttributeUpdate]

class UpdateFriendDetailsResponse(BaseModel):
    friend_name: str
    attributes: List[FriendAttributeUpdate]

File: front/src/interfaces/friend.ts
export interface Friend {
  name: string;
  user_id: number;
  id: number;
}

export interface FriendAttribute {
  attribute_name: string;
  value: string;
}

export interface Conversation {
  context: string;
  conversation_date: string;
}

export interface FriendDetails {
  friend_name: string;
  attributes: FriendAttribute[];
  conversations: Conversation[];
}
